### PR TITLE
[CAS-1428] Fixing videos attachments for video, audio and PDFs

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
[CAS-1428](https://stream-io.atlassian.net/browse/CAS-1428)
### 🎯 Goal

This PR fixes the handle of videos, audios and PDFs from other SDKs. Currently videos  and audios from Flutter and iOS were not displayable in our SDK, also PDFs with spaces in the names fail to load.

### 🛠 Implementation details

- Using type of Attachment as a fallback if mime_type is not supported and adding more mime_types to conform with what we receive from iOS and Flutter.
- Removing spaces and unwanted characters from name of files

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/10619102/141131669-4924c380-31f9-4cf1-87df-9738c88fdf2d.mov" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/10619102/141131641-68b28c94-d336-4580-8d96-00c7c8b29f29.mov" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### Testing

Upload this branch to local maven and test it in this project: https://github.com/GetStream/AttachmentsSample. Observe that all types of video how as expected. 

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes.
- ~[ ] New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes.
- ~[ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))~
- [x] Reviewers added.

### 🎉 GIF
giphy

### CODE REVIEW

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/ModelType.kt [hunk 1]
```diff
@@ -49,7 +49,10 @@ public object ModelType {
     public const val attach_mime_xlsx: String = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
     public const val attach_mime_zip: String = "application/zip"
     public const val attach_mime_mov: String = "video/mov"
-    public const val attach_mime_mp4: String = "video/mp4"
+    public const val attach_mime_quicktime: String = "quicktime"
+    public const val attach_mime_video_quicktime: String = "video/quicktime"
+    public const val attach_mime_mp4: String = "mp4"
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/ModelType.kt:54]: This is the mime type from Flutter for MP4

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/ModelType.kt [hunk 2]
```diff
@@ -49,7 +49,10 @@ public object ModelType {
     public const val attach_mime_xlsx: String = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
     public const val attach_mime_zip: String = "application/zip"
     public const val attach_mime_mov: String = "video/mov"
-    public const val attach_mime_mp4: String = "video/mp4"
+    public const val attach_mime_quicktime: String = "quicktime"
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/ModelType.kt:52]: mime_type from Flutter for .MOV

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/ModelType.kt [hunk 3]
```diff
@@ -49,7 +49,10 @@ public object ModelType {
...
+    public const val attach_mime_video_quicktime: String = "video/quicktime"
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/ModelType.kt:53]: mime_type from iOS for .MOV

**Review by samiuelson — APPROVED** [review]

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/ModelType.kt (hunk 2 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/ModelType.kt:53]: Is this full mime type? That's odd. Should be `video/quicktime`.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/navigation/destinations/AttachmentDestination.kt [hunk 4]
```diff
@@ -22,13 +22,16 @@ public open class AttachmentDestination(
...
+    private val logger = ChatLogger.get("AttachmentDestination")
+
     override fun navigate() {
         showAttachment(message, attachment)
     }
 
     public fun showAttachment(message: Message, attachment: Attachment) {
         if (attachment.type == ModelType.attach_file ||
             attachment.type == ModelType.attach_video ||
+            attachment.type == ModelType.attach_audio ||
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/navigation/destinations/AttachmentDestination.kt:34]: What about us? We still continue to send audio attachments as files? And therefore they are not playable on IOS?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/navigation/destinations/AttachmentDestination.kt [hunk 5]
```diff
@@ -135,5 +167,10 @@ public open class AttachmentDestination(
 
     private companion object {
         private const val VIDEO_MIME_TYPE_PREFIX = "video"
+        private const val MP4_MIME_TYPE = "mp4"
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/navigation/destinations/AttachmentDestination.kt:170]: Are they going to fix that on their side? `mp4` is not a mime type , it is mime subtype.

**Review by bychkovdmitry — APPROVED** [review]

**Comment by bychkovdmitry** [comment]
Happy that somebody finally made this not so pleasant task. Thanks! 👍

**Review by samiuelson — APPROVED** [review]

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**